pkg/controller/controllercmd/cmd.go

```diff
@@ -83,10 +85,14 @@ func (c *ControllerCommandConfig) NewCommand() *cobra.Command {
 // The context passed will be passed down to controller loops and observers and cancelled on SIGTERM and SIGINT signals.
 func (c *ControllerCommandConfig) NewCommandWithContext(ctx context.Context) *cobra.Command {
 	cmd := &cobra.Command{
-		Run: func(cmd *cobra.Command, args []string) {
+		RunE: func(cmd *cobra.Command, args []string) error {
 			// boiler plate for the "normal" command
 			rand.Seed(time.Now().UTC().UnixNano())
-			logs.InitLogs()
+
+			// Activate logging as soon as possible.
+			if err := c.logs.ValidateAndApply(); err != nil {
```

Comment:
Why not using the same pattern as below with `klog.Fatal`? The result is identical, we fail.

Reply:
:facepalm: of course we don't have klog setup, so print error and `os.Exit(1)`

Reply:
reverted back to `Run`